55 visualize evaluation metrics

benathanael opened the pull request:
## What does this PR do?

This PR introduces a new script for visualizing benchmark metrics as a function of task parameters (e.g. tau_m, lambda_val).

Main changes:
- **`src/utils/plot_metric_vs_taskparam.py`**: New CLI script for generating line plots of metrics vs. task parameters.  
- **`src/utils/LinePlot.py`**: Updated to support plotting against any numeric column from `metrics.csv` (not just `num_simulations`).  
- **`src/tests/LinePlot_demo.py`** + **`tests/dummy_csv_4x4_grid_with_tau.csv`**: Demo showing how to use `LinePlot` with an additional task parameter column (e.g., `tau_m`).  
- **`tests/test_plot_metric_vs_taskparam.py`**: Simple unit test that checks whether a `.png` file is created in the correct folder `outputs/{task}_{method}/plots/`.

This enables future benchmarks to save task parameters directly into `.csv` and visualize them consistently.  

## Does this close any issues?

Closes #55  

## Any relevant code examples, logs, or error messages?

Example usage:  
Input:

```bash
python -m src.utils.plot_metric_vs_taskparam \
  --input_path tests/dummy_csv_1x1_grid_with_tau.csv \
  --metric C2ST \
  --task_param tau_m
```

Output:

<img width="1530" height="912" alt="CleanShot 2025-08-22 at 13 45 35@2x" src="https://github.com/user-attachments/assets/3bbbc72f-e086-4588-b0d8-7eb52b35be91" />


## ✅ Checklist

Put an `x` in the boxes that apply. If you're unsure about any of them, no worries—just ask!

- [x] I have read and followed the [contribution guidelines](https://github.com/sbi-dev/sbi/blob/main/CONTRIBUTING.md)
- [x] I have added helpful comments to my code where needed
- [x] I have added tests for new functionality
- [ ] (If applicable) I have reported how long new tests run and marked them with `pytest.mark.slow`

**For reviewers:**
- [ ] I have reviewed every file
- [ ] All comments have been addressed

benathanael committed:
feat(LinePlot): generalize x-axis to any metrics.csv column

Previously, LinePlot only supported plotting metrics as a function of num_simulations.
This update generalizes the x-axis to accept any numeric column from metrics.csv.

For demonstration purposes, I also added a dummy CSV with arbitrary tau and C2ST values, along with a script under tests/LinePlot_demo.py to show how the plot looks with the new functionality.

benathanael committed:
feat(plotting): add plot_metric_vs_taskparam script

Adds a new script `plot_metric_vs_taskparam.py` for generating line plots of
evaluation metrics against task parameters. Includes a unit test
`test_plot_metric_vs_taskparam.py` to ensure correct plot creation, and a
dummy CSV (`dummy_csv_1x1_grid_with_tau.csv`) for trying out the script
manually.

jmktpl reviewed (changes requested):
Thank you for adding a test to check if a .png was created! I think this is a good option to test the visualization.

I also tested the visualization with the Demo file you provided. It works, but i had to adjust the paths in `tests/LinePlot_demo.py` to e.g. "tests/dummy_csv_4x4_grid_linear.csv", so that I can run the code with `python tests/LinePlot_demo.py`

Would be nice, if you could change it real quick, as it is a nice check for a more complex visualization. :)

Except for this: Looks good to me! :)

benathanael committed:
fix(csv_utils): resolve relative data_sources against base_directory

Previously gather_csv_files only joined base_directory for glob patterns.
Relative file and directory paths were still interpreted against the terminal CWD,
causing demos to fail when run outside the project root.

This change normalizes all relative sources against base_directory,
ensuring consistent behavior for files, directories, and glob patterns.

benathanael committed:
fix(LinePlot_demo): ensure demo runs correctly from any working directory

Now explicitly passes the script directory as the base directory in LinePlot(). This ensures the demo works regardless of the current working directory.

benathanael commented:
Thank you for your testing and your suggestion!  

Changing the paths from `dummy_[...].csv` to `tests/dummy_[...].csv` made the script runnable when the working directory was the **project root** (e.g. when running `python -m tests.LinePlot_demo`), but it failed when the working directory was `tests/` (e.g. when running directly in the IDE). To support both cases, I set `base_directory` in the `LinePlot()` calls to the **script’s directory**.  

While doing this, I also fixed a bug in `gather_csv_files(...)` in `utils/csv_utils`, where relative directories were not resolved against `base_directory`.  

With these changes, the script now runs consistently from any working directory, which should solve your described issue!

jmktpl commented:
Great, Thank you! Now it works for me!
